GLOSSARY.md

```diff
@@ -15,3 +15,22 @@ https://facebook.github.io/react/
 
 # Ember
 http://emberjs.com/
+
+# Leader system
+The sub-system acting as the authority for the data in question. It encapsulates all logic for this area of business capability, and is responsible for providing suitable access to all authorized consumers.
+
+# Follower system
+A sub-system which aggregates information from multiple leaders. A reporting system might, for example, maintain useful representations of data that cross many leader systems. JOINing customer-supplied metadata with fundraising page performance is a good example of this pattern.
+
+# OLTP
+On line transaction processing, or OLTP, is a class of information systems that facilitate and manage transaction-oriented applications, typically for data entry and retrieval transaction processing.
+
+# OLAP
+OLAP (online analytical processing) is computer processing that enables a user to easily and selectively extract and view data from different points of view. For example, a user can request that data be analyzed to display a spreadsheet showing all of a company's beach ball products sold in Florida in the month of July, compare revenue figures with those for the same products in September, and then see a comparison of other product sales in Florida in the same time period. To facilitate this kind of analysis, OLAP data is stored in a multidimensional database. Whereas a relational database can be thought of as two-dimensional, a multidimensional database considers each data attribute (such as product, geographic sales region, and time period) as a separate "dimension." OLAP software can locate the intersection of dimensions (all products sold in the Eastern region above a certain price during a certain time period) and display them. Attributes such as time periods can be broken down into subattributes.
+
+# Semantic event
+A business-domain event, with rich information that should be understandable by any person familiar with our day-to-day business operations. For example, 'Penny Smith created a supporter page for 40HF.'
```

Comment:
It might just be my ES background but I don't think there is a requirement for semantic events to be understandable "by any person familiar with our day-to-day business operations" as they will be tied to a specific domain (e.g. payments). I do agree that they **are** business domain events, understandable by anyone with that domain knowledge and we should work towards publishing documentation that exposes those domain concepts.

An example set of hypothetical semantic events from an onboarding application:

* CharityCreated(charity_id, display_name, dba)
* CharityCommencedKYC(charity_id, commenced_at)
* CharityCompletedKYC(charity_id, completed_at, approved_by)
* CharityCommencedOnboarding(charity_id, commenced_at)
* CharityCompletedOnboarding(charity_id, completed_at, approved_by)

Only someone with domain knowledge understands what each of these events relate to and when it is appropriate to raise them but anyone familiar with our day to day operations would know that there is an onboarding process, not the finer details.

Because of this Payments has been loosely throwing around the concept of "synthetic events" which are non-semantic but contain a rolled up and flattened view of of the semantic events for reporting / analytic use which is very similar to what you have described.

If we were representing those events as synthetic events I would envision something like the following:

* CharityCreated(charity_id, display_name, dba)
* CharityCompletedKYC(charity_id, display_name, dba, kyc_status)
* CharityCompletedOnboarding(charity_id, display_name, dba, kyc_status, onboarding_status)

These events summarise and flatten the domain's semantic events into higher level, potentially more relevant events (almost serialising current state).

Maybe I'm reading too much into it and we only have to make this distinction in ES systems where we derive current state from events?